src/Servers/Kestrel/Transport.Quic/test/WebHostTests.cs

```diff
@@ -0,0 +1,212 @@
+                                await context.Response.WriteAsync("hello, world");
+                            });
+                        });
+                })
+                .ConfigureServices(AddTestLogging);
+
+            using (var host = builder.Build())
+            using (var client = new HttpClient())
+            {
+                await host.StartAsync();
+
+                var request = new HttpRequestMessage(HttpMethod.Get, $"https://127.0.0.1:{host.GetPort()}/");
+                request.Version = HttpVersion.Version30;
+                request.VersionPolicy = HttpVersionPolicy.RequestVersionExact;
+
+                // Act
+                var response = await client.SendAsync(request);
+
+                // Assert
+                response.EnsureSuccessStatusCode();
+                Assert.Equal(HttpVersion.Version30, response.Version);
+                var responseText = await response.Content.ReadAsStringAsync();
+                Assert.Equal("hello, world", responseText);
+
+                await host.StopAsync();
+            }
+        }
+
+        [ConditionalTheory]
+        [MsQuicSupported]
+        [InlineData(5002, 5003)]
+        [InlineData(5004, 5004)]
+        public async Task Listen_Http3AndSocketsCoexistOnDifferentEndpoints_ClientSuccess(int http3Port, int http1Port)
+        {
+            // Arrange
+            var builder = GetHostBuilder()
+                .ConfigureWebHost(webHostBuilder =>
+                {
+                    webHostBuilder
+                        .UseKestrel(o =>
+                        {
+                            o.Listen(IPAddress.Parse("127.0.0.1"), http3Port, listenOptions =>
+                            {
+                                listenOptions.Protocols = Core.HttpProtocols.Http3;
+                                listenOptions.UseHttps();
+                            });
+                            o.Listen(IPAddress.Parse("127.0.0.1"), http1Port, listenOptions =>
+                            {
+                                listenOptions.Protocols = Core.HttpProtocols.Http1;
+                                listenOptions.UseHttps();
+                            });
+                        })
+                        .Configure(app =>
+                        {
+                            app.Run(async context =>
+                            {
+                                await context.Response.WriteAsync("hello, world");
+                            });
+                        });
+                })
+                .ConfigureServices(AddTestLogging);
+
+            using var host = builder.Build();
+            await host.StartAsync().DefaultTimeout();
+
+            await CallHttp3AndHttp1EndpointsAsync(http3Port, http1Port);
+
+            await host.StopAsync().DefaultTimeout();
+        }
+
+        [ConditionalFact]
+        [MsQuicSupported]
+        public async Task Listen_Http3AndSocketsCoexistOnSameEndpoint_ClientSuccess()
```

Comment:
Add an alt-svc test that negotiates up to HTTP/2 from HTTP/1 or HTTP/2.

Reply:
I can't do that at the moment because HTTP/1.1 and HTTP/2 require suppression of cert error using callback, and HTTP/3 errors when there is a callback.

I'll create an issue to add this test.

Reply:
https://github.com/dotnet/aspnetcore/issues/34148